controller/tests/controller_server/utils_test.py

```diff
@@ -383,51 +385,71 @@ def test_get_volume_id_info_no_id_fail(self):
 
+    def _check_node_id_parameters(self, node_id_info, nvme_nqn, fc_wwns, iscsi_iqn):
+        self.assertEqual(node_id_info.node_name, "host-name")
+        self.assertEqual(node_id_info.initiators.nvme_nqn, nvme_nqn)
+        self.assertEqual(node_id_info.initiators.fc_wwns, fc_wwns.split(":"))
+        self.assertEqual(node_id_info.initiators.iscsi_iqn, iscsi_iqn)
+
     def test_get_node_id_info(self):
         with self.assertRaises(array_errors.HostNotFoundError) as ex:
-            utils.get_node_id_info("badnodeformat")
+            utils.get_node_id_info("bad-node-format")
             self.assertTrue("node" in str(ex))
+        host_name = "host-name"
+        nvme_nqn = "nqn.ibm"
+        fc_wwns = "wwn1:wwn2"
+        iscsi_iqn = "iqn.ibm"
 
-        hostname, fc_wwns, iscsi_iqn = utils.get_node_id_info("hostabc;;iqn.ibm")
-        self.assertEqual(hostname, "hostabc")
-        self.assertEqual(iscsi_iqn, "iqn.ibm")
-        self.assertEqual(fc_wwns, "")
+        node_id_info = NodeIdInfo("{};;;{}".format(host_name, iscsi_iqn))
+        self._check_node_id_parameters(node_id_info, "", "", iscsi_iqn)
 
-        hostname, fc_wwns, iscsi_iqn = utils.get_node_id_info("hostabc;wwn1:wwn2;iqn.ibm")
-        self.assertEqual(hostname, "hostabc")
-        self.assertEqual(iscsi_iqn, "iqn.ibm")
-        self.assertEqual(fc_wwns, "wwn1:wwn2")
+        node_id_info = NodeIdInfo("{};;{};{}".format(host_name, fc_wwns, iscsi_iqn))
+        self._check_node_id_parameters(node_id_info, "", fc_wwns, iscsi_iqn)
 
-        hostname, fc_wwns, iscsi_iqn = utils.get_node_id_info("hostabc;wwn1:wwn2")
-        self.assertEqual(hostname, "hostabc")
-        self.assertEqual(iscsi_iqn, "")
-        self.assertEqual(fc_wwns, "wwn1:wwn2")
+        node_id_info = NodeIdInfo("{};{};{}".format(host_name, nvme_nqn, fc_wwns))
+        self._check_node_id_parameters(node_id_info, nvme_nqn, fc_wwns, "")
 
-    def test_choose_connectivity_types(self):
-        res = utils.choose_connectivity_type(["iscsi"])
-        self.assertEqual(res, "iscsi")
+        node_id_info = NodeIdInfo("{};{}".format(host_name, nvme_nqn))
+        self._check_node_id_parameters(node_id_info, nvme_nqn, "", "")
 
-        res = utils.choose_connectivity_type(["fc"])
-        self.assertEqual(res, "fc")
+        node_id_info = NodeIdInfo("{};;{}".format(host_name, fc_wwns))
+        self._check_node_id_parameters(node_id_info, "", fc_wwns, "")
 
-        res = utils.choose_connectivity_type(["iscsi", "fc"])
-        self.assertEqual(res, "fc")
+    def test_choose_connectivity_types(self):
+        nvme = NVME_OVER_FC_CONNECTIVITY_TYPE
+        fc = FC_CONNECTIVITY_TYPE
+        iscsi = ISCSI_CONNECTIVITY_TYPE
+        expected_chosen_by_connectivities_found = {(nvme, fc, iscsi): nvme, (fc, iscsi): fc, (nvme,): nvme, (fc,): fc,
+                                                   (iscsi,): iscsi
+                                                   }
```

Comment:
is this the best indentation/formatting?
```suggestion
        expected_chosen_by_connectivities_found = {
            (nvme, fc, iscsi): nvme, (fc, iscsi): fc,
            (nvme,): nvme, (fc,): fc, (iscsi,): iscsi
        }
```

Reply:
done